core/primitives-core/src/profile.rs

```diff
@@ -1,13 +1,51 @@
 use std::fmt;
+use std::ops::{Index, IndexMut};
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use serde::{Deserialize, Serialize};
 
 use crate::config::{ActionCosts, ExtCosts};
-use crate::types::Gas;
 
-/// Data for total gas burnt (index 0), and then each external cost and action
-type DataArray = [u64; ProfileData::LEN];
+#[derive(Clone, PartialEq, Eq)]
+pub struct DataArray([u64; Self::LEN]);
```

Comment:
```suggestion
pub struct DataArray(Box<[u64; Self::LEN]>);
```

This struct has rather big size_of, so it's going to be faster to box this once,  and then pass the pointer around.